presto-parquet/src/main/java/com/facebook/presto/parquet/reader/ParquetReader.java

```diff
@@ -408,4 +515,77 @@ private static Block rewriteLongArrayBlock(LongArrayBlock longArrayBlock, Type t
 
         return newBlockBuilder.build();
     }
+
+    private static <T> List<T> listWithNulls(int size)
+    {
+        return Stream.generate(() -> (T) null).limit(size).collect(Collectors.toCollection(ArrayList<T>::new));
+    }
+
+    private RowRanges getRowRanges(int blockIndex)
+    {
+        assert filter != null;
+
+        RowRanges rowRanges = blockRowRanges.get(blockIndex);
+        if (rowRanges == null) {
+            rowRanges = ColumnIndexFilter.calculateRowRanges(FilterCompat.get(filter), blockIndexStores.get(blockIndex),
+                    paths.keySet(), blocks.get(blockIndex).getRowCount());
+            blockRowRanges.set(blockIndex, rowRanges);
+        }
+        return rowRanges;
+    }
+
+    /**
+     * Describes a list of consecutive parts to be read at once. A consecutive part may contain whole column chunks or
+     * only parts of them (some pages).
+     */
+    private class ConsecutivePartList
```

Comment:
s/ConsecutivePartList/PageRanges/g

Reply:
get it. We could either add in comments above, describing the private class is from Parquet, or rename it to PageRanges

Reply:
I just replace it with PageRanges